wifi: SAP mode docs (#13562)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf70/developing/sap.rst

.. _nRF70_soft_ap_mode:

Soft AP mode
############

.. contents:: Table of Contents
   :local:
   :depth: 2

The nRF70 Series device supports Software-enabled Access Point ( :abbr:`SAP (Software-enabled Access Point)` ) mode.
This mode allows a device to act as a virtual router or temporary access point (AP) for other station devices to connect to it.
This is typically used in scenarios where you want to share the device's internet connection with other station devices.
For example, when you set up a hotspot on your smartphone, it's using SAP mode.
The smartphone acts as a virtual router, allowing other station devices to connect to the internet through it.

The nRF70 Series devices use SAP mode to support Wi-Fi based provisioning.

> The SAP mode support is experimental and is not recommended for production use.

Reviewer: Do we need to state this here ? I guess this should be updated in https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13562/nrf/releases_and_maturity/software_maturity.html#wi-fi-feature-support ?

Reviewer 2: To avoid redundancy, I suggest at least mentioning "The SAP mode support is experimental" and linking to software_maturity.rst under "experimental", as we do in other places.

Author: ok, I forgot to tag this support as `EXPERIMENTAL` will do it in this PR.

Features
********

The SAP mode supports below features in the nRF70 Series devices:

>   * An interface can operate either in SAP mode or Station mode, but not both at the same time.

Author: Zephyr only has a single interface type, I can change to `network interface`

>   * The SAP mode supports only Wi-Fi 4 (802.11n) only.

Author: We are not providing any configuration, it *only* operates in `11n`.

>   * The SAP mode supports only Wi-Fi 4 (802.11n) only.

Reviewer 3: OK

>   * Primarily intended for a single client connection.

Reviewer 3: MAybe you can add a more strict sentence. E.g.
The SAP mode current supports up to a single Station device connection

>   * Primarily intended for a single client connection.

Author: Supports up to 8, but tested only with 1, so, may be: `SAP mode support up to eight clients, but the testing has been carried out with a single client`?

  * Wi-Fi personal security (WPA2-PSK) is only supported.

>   * Minimal performance needed for Wi-Fi based provisioning.

Author: ok, will remove

> SAP mode enablement

Reviewer 3: ```suggestion
Enabling the SAP mode
```

*******************

> The SAP mode is enabled by setting the :kconfig:option:`CONFIG_NRF700X_AP_MODE` and :kconfig:option:`CONFIG_WPA_SUPP_AP` options in the Zephyr configuration file.

Reviewer 3: ```suggestion
The SAP mode is enabled by setting the :kconfig:option:`CONFIG_NRF700X_AP_MODE` and :kconfig:option:`CONFIG_WPA_SUPP_AP` options in the application project configuration file.
```

SAP mode configuration
**********************

> The SAP mode is configured using the standard Wi-Fi configuration APIs in the Zephyr networking stack.

Reviewer 3: IS there a handy link to that? Would be nice to have it here :)

     * - API
       - Description
       - Parameters
     * - ``NET_REQUEST_WIFI_AP_ENABLE``
       - Starts the SAP mode.
       - | Interface
         | SSID
         | Channel
         | Security information
     * - ``NET_REQUEST_WIFI_AP_DISABLE``
       - Stops the SAP mode.
       - Interface
     * - ``NET_REQUEST_WIFI_STATUS``
       - Displays the status of the SAP mode interface.
       - Interface

The SAP mode generates the following events:

.. list-table::
     :header-rows: 1

     * - Event
       - Description
     * - ``NET_EVENT_WIFI_AP_ENABLE_RESULT``
       - Generated when the SAP mode is enabled.
     * - ``NET_EVENT_WIFI_AP_DISABLE_RESULT``
       - Generated when the SAP mode is disabled.
     * - ``NET_EVENT_WIFI_AP_STA_CONNECTED``
       - Generated when a station connects to the SAP mode interface.
     * - ``NET_EVENT_WIFI_AP_STA_DISCONNECTED``
       - Generated when a station disconnects from the SAP mode interface.

.. note::
   There is no API to display the list of stations connected to the SAP mode interface.
   But using the events generated by Zephyr Wi-Fi stack, you can get the list of stations connected to the SAP mode interface.

.. note::

>     The SAP mode operation is governed by regulatory requirements.

Reviewer 3: ```suggestion
    The SAP mode operation is dictated by regulatory requirements.
```

    For more information on regulatory support, see :ref:`ug_nrf70_developing_regulatory_support`.

Stations handling
*****************

For a typical SAP mode operation, the nRF70 Series device acts as a virtual router, allowing other station devices to connect to it.
The SAP interface needs to be configured with a valid IP address and a subnet mask.
But the station devices use :abbr:`DHCP (Dynamic Host Configuration Protocol)` to get an IP address from the virtual router.

> The Zephyr networking stack does not support DHCP server functionality.

Reviewer 3: ```suggestion
Currently the Zephyr networking stack does not support DHCP server functionality.
```
Also, could this be made into a note?

.. _nRF70_soft_ap_mode:

SoftAP mode
###########

.. contents::
   :local:
   :depth: 2

The nRF70 Series device supports :term:`Software-enabled Access Point (SoftAP or SAP)` mode.
This mode allows a device to operate as a virtual router or temporary access point (AP) and accept connections from other Wi-Fi station devices.
SoftAP mode is typically used in scenarios where a Wi-Fi device wishes to share its internet connection with nearby station devices.
For instance, smartphone Wi-Fi hotspots utilize SoftAP mode.
The smartphone acts as a virtual router, allowing other station devices to connect to the internet through it.
Alternatively, SoftAP mode can be used to securely provision Wi-Fi devices into an access point network.

nRF70 Series devices support Wi-Fi provisioning using the SoftAP mode as one of the provisioning methods.

Supported functionality and limitations
***************************************

The SAP mode in the nRF70 Series devices provides SoftAP support with the following limitations:

>   * A network interface can operate in either SAP mode or Station mode, but not in both modes simultaneously.
>   * SAP mode only supports the Wi-Fi 4 (IEEE 802.11n) mode of operation.
>   * The SAP mode currently supports a single station device connection.
> 
>       However, the implementation provides an :ref:`experimental <software_maturity>` support for up to eight station connections.
> 
>   * Only Wi-Fi personal security (WPA2-PSK) is supported.

Reviewer 4: Formatting issue https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-13562/nrf/device_guides/working_with_nrf/nrf70/developing/sap.html#supported-functionality-and-limitations
```suggestion
  * A network interface can operate in either SAP mode or Station mode, but not in both modes simultaneously.
  * SAP mode only supports the Wi-Fi 4 (IEEE 802.11n) mode of operation.
  * The SAP mode currently supports a single station device connection.
     However, the implementation provides an :ref:`experimental <software_maturity>` support for up to eight station connections.
 * Only Wi-Fi personal security (WPA2-PSK) is supported.
```

Author: if I don't put a newline, the previous line is highlighted in `bold`

.. _nRF70_soft_ap_mode:

SoftAP mode
###########

.. contents:: Table of Contents
   :local:
   :depth: 2

The nRF70 Series device supports Software-enabled Access Point ( :abbr:`SAP (Software-enabled Access Point)` ) mode.
This mode allows a device to operate as a virtual router or temporary access point (AP) and accept connections from other Wi-Fi Station devices.
SoftAP mode is typically used in scenarios where a Wi-Fi device wishes to share its internet connection with nearby station devices.
Smartphone Wi-Fi hotspots, for example, utilize the SoftAP mode.

> The smartphone acts as a virtual router, allowing other station devices to connect to the internet through it.

Reviewer 3: ```suggestion
The smartphone acts as a virtual router, allowing other station devices to connect to the internet through it.
SoftAP mode may, alternatively, be used to securely provision Wi-Fi devices into an access point network.
```

> The nRF70 Series devices use SAP mode to support Wi-Fi based provisioning.

Reviewer 3: ```suggestion
nRF70 Series devices support Wi-Fi provisioning using the SoftAP mode.
```

Reviewer 3: Note to @greg-fer : I hope the sentence above is not too "strong", i.e. it does not imply that SoftAP-based provisioning is the _only_ way to provision nRF70 Series devices.

Author: I have added `as one of the provisioning methods.`

Supported functionality and limitations
***************************************

The SAP mode in the nRF70 Series devices provides SoftAP support with the following limitations:

  * A network interface can operate either in SAP mode or Station mode, but not both at the same time.

>   * The SAP mode supports only Wi-Fi 4 (802.11n) only.

Reviewer 3: ```suggestion
  * The SAP mode supports only Wi-Fi 4 (IEEE 802.11n) mode of operation.
```

>   * The SAP mode supports up to 8 stations, but primary testing has been done with only a single station.

Reviewer 3: How about:

```suggestion
  * The SAP mode currently supports a single station device connection. The implementation provides _experimental_ support for up to 8 station connection is added.
```

     * - API
       - Description
       - Parameters
     * - ``NET_REQUEST_WIFI_AP_ENABLE``
       - Starts the SAP mode.
       - | Interface
         | SSID
         | Channel
         | Security information
     * - ``NET_REQUEST_WIFI_AP_DISABLE``
       - Stops the SAP mode.
       - Interface
     * - ``NET_REQUEST_WIFI_STATUS``
       - Displays the status of the SAP mode interface.
       - Interface

The SAP mode generates the following events:

.. list-table::
     :header-rows: 1

     * - Event
       - Description
     * - ``NET_EVENT_WIFI_AP_ENABLE_RESULT``
       - Generated when the SAP mode is enabled.
     * - ``NET_EVENT_WIFI_AP_DISABLE_RESULT``
       - Generated when the SAP mode is disabled.
     * - ``NET_EVENT_WIFI_AP_STA_CONNECTED``
       - Generated when a station connects to the SAP mode interface.
     * - ``NET_EVENT_WIFI_AP_STA_DISCONNECTED``
       - Generated when a station disconnects from the SAP mode interface.

.. note::
   There is no API to display the list of stations connected to the SAP mode interface.
   But using the events generated by Zephyr Wi-Fi stack, you can get the list of stations connected to the SAP mode interface.

> .. note::

Author: But both are different topics, let's leave it as is.

> .. note::

Reviewer 4: May be then it could be written as a plain text?

> .. note::

Author: No, but this is important to highlight.
